pkg/agent/supportbundlecollection/support_bundle_controller_test.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package supportbundlecollection
+
+import (
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stesting "k8s.io/client-go/testing"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/exec"
+
+	agentquerier "antrea.io/antrea/pkg/agent/querier"
+	"antrea.io/antrea/pkg/apis/controlplane"
+	cpv1b2 "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+	"antrea.io/antrea/pkg/client/clientset/versioned"
+	fakeversioned "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	"antrea.io/antrea/pkg/ovs/ovsctl"
+	"antrea.io/antrea/pkg/querier"
+	"antrea.io/antrea/pkg/support"
+)
+
+type fakeController struct {
+	*SupportBundleController
+	mockController *gomock.Controller
+}
+
+type antreaClientGetter struct {
+	clientset versioned.Interface
+}
+
+func (g *antreaClientGetter) GetAntreaClient() (versioned.Interface, error) {
+	return g.clientset, nil
+}
+
+func newFakeController(t *testing.T) (*fakeController, *fakeversioned.Clientset) {
+	controller := gomock.NewController(t)
+	clientset := &fakeversioned.Clientset{}
+	supportBundleController := NewSupportBundleController("vm1", controlplane.SupportBundleCollectionNodeTypeExternalNode, "vm-ns", &antreaClientGetter{clientset}, nil,
+		nil, nil, true, true)
+	supportBundleController.sftpUploader = &testUploader{}
+	return &fakeController{
+		SupportBundleController: supportBundleController,
+		mockController:          controller,
+	}, clientset
+}
+
+func TestSupportBundleCollectionAdd(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		supportBundleCollection *cpv1b2.SupportBundleCollection
+	}{
+		{
+			name:                    "Add SupportBundleCollection",
+			supportBundleCollection: generateSupportbundleCollection("supportBundle1", "sftp://10.220.175.92:22/root/supportbundle"),
+		},
+		{
+			name:                    "Add SupportBundleCollection without url prefix",
+			supportBundleCollection: generateSupportbundleCollection("supportBundle2", "10.220.175.92:22/root/supportbundle"),
+		},
```

Comment:
Maybe we could add a failure case and add a "expectedCompleted"?

Reply:
Thanks for the suggestion. I have refactored this part, please help check the latest version.